cpp/src/lists/drop_list_duplicates.cu

```diff
@@ -462,22 +572,23 @@ std::unique_ptr<column> drop_list_duplicates(lists_column_view const& lists_colu
                                              rmm::mr::device_memory_resource* mr)
 {
   if (lists_column.is_empty()) return cudf::empty_like(lists_column.parent());
-  if (cudf::is_nested(lists_column.child().type())) {
-    CUDF_FAIL("Nested types are not supported in drop_list_duplicates.");
+  if (auto const child_type = lists_column.child().type();
+      cudf::is_nested(child_type) && child_type.id() != type_id::STRUCT) {
+    CUDF_FAIL("Nested types other than STRUCT are not supported in `drop_list_duplicates`.");
   }
 
-  // Flatten all entries (depth = 1) of the lists column
+  // Flatten all entries (depth = 1) of the lists column.
   auto const lists_entries = lists_column.get_sliced_child(stream);
 
-  // sorted_lists will store the results of the original lists after calling segmented_sort
+  // sorted_lists will store the results of the original lists after calling segmented_sort.
   auto const sorted_lists = [&]() {
     // If nans_equal == ALL_EQUAL and the column contains lists of floating-point data type,
-    // we need to replace -NaN by NaN before sorting
+    // we need to replace -NaN by NaN before sorting.
     auto const replace_negative_nan =
-      nans_equal == nan_equality::ALL_EQUAL and
-      type_dispatcher(lists_entries.type(), detail::has_negative_nans_fn{}, lists_entries, stream);
+      nans_equal == nan_equality::ALL_EQUAL &&
+      type_dispatcher(
+        lists_entries.type(), detail::has_negative_nans_dispatch{}, lists_entries, stream);
```

Comment:
It seems redundant to first check if negative nan values exist before launching a kernel to replace negative nans. In the case where negative NaNs exist, it would be faster to just always run NaN replacement kernel. In the case where NaN's _don't_ exist, I believe unnecessarily running the NaN replacement function will be just as fast as checking if NaNs exist.

Reply:
Also, is replacing the NaNs necessary? In the regular `cudf::sort` we already treat all NaNs as equivalent in the row comparator, so there isn't a need there to do any NaN replacement.

Reply:
Yeah, but there is one issue. If we call replace NaN, we will do copying---materialize a new column (which may be nested) and copy data from the old column over it. The result will always be a new column regardless of whether it has NaN or not. I try to avoid that by probing NaN before calling to replacing NaN.

Reply:
For large columns (nested---structs columns with many members), I believe that avoiding unnecessary copy is beneficial.

Reply:
Oh, I was thinking we could do the replace NaN operation in place.